server/jetstream_cluster.go

```diff
@@ -3452,8 +3452,6 @@ func (js *jetStream) processStreamAssignment(sa *streamAssignment) bool {
 		return false
 	}
 
-	var didRemove bool
```

Comment:
`didRemove` used to be set a long time ago, but now it would always be `false`.

To ensure the meta snapshot is written both for creation and removal, return `true` at the end instead.